### ISSUE TITLE
Stalling state sync, dropping peer

### ISSUE THREAD

**Issue by MitchK** [ethereum/go-ethereum#15576]
#### System information

Geth version:  1.7.3
OS & Version: Ubuntu (Docker)

#### Expected behaviour

Should sync up using fast sync

#### Actual behaviour

```
...
INFO [11-28|23:29:33] Imported new state entries               count=0    elapsed=532.808µs processed=3981896 pending=17    retry=16  duplicate=0 unexpected=768
WARN [11-28|23:31:05] Stalling state sync, dropping peer       peer=6fea8f23eaa1b5f6
WARN [11-28|23:34:02] Stalling state sync, dropping peer       peer=8df0737b5af6fabc
INFO [11-28|23:34:30] Imported new state entries               count=263  elapsed=9.078µs   processed=3982159 pending=3999  retry=0   duplicate=0 unexpected=768
WARN [11-28|23:34:37] Stalling state sync, dropping peer       peer=2f11578ec215105d
WARN [11-28|23:34:43] Stalling state sync, dropping peer       peer=e2f7c77207d76057
WARN [11-28|23:40:46] Stalling state sync, dropping peer       peer=6788d56813b3549e
INFO [11-28|23:41:07] Imported new state entries               count=384  elapsed=8.215µs   processed=3982543 pending=10044 retry=2   duplicate=0 unexpected=768
WARN [11-28|23:41:20] Stalling state sync, dropping peer       peer=f41c361a0a27af8b
WARN [11-28|23:41:38] Stalling state sync, dropping peer       peer=16a47bd198cfe1cf
WARN [11-28|23:44:10] Stalling state sync, dropping peer       peer=2f11578ec215105d
WARN [11-28|23:47:52] Stalling state sync, dropping peer       peer=6fea8f23eaa1b5f6
WARN [11-28|23:47:55] Stalling state sync, dropping peer       peer=aa5ee49c3a44f337
```

#### Steps to reproduce the behaviour

Docker + Amazon EBS (SSD, gp2)

```
geth --testnet --ipcdisable --rpc --rpcaddr 0.0.0.0 --rpcport 8545 --datadir ./.ethereum --rpcapi eth,web3,net --verbosity 3 --syncmode fast
```

**Comment by pzagor2** [comment]
I have the exact same issue. I'm running an instance on Scaleway - Ubuntu Yakkety (16.10 latest) -
 Geth - v1.6.7-stable-ab5646c5

currentBlock is stuck and does not move forward.
```
> web3.eth.syncing
{
  currentBlock: 4369910,
  highestBlock: 4370001,
  knownStates: 9300,
  pulledStates: 4028,
  startingBlock: 4369864
}
```

**Comment by pzagor2** [comment]
@MitchK Check what is your `geth` version. I followed instructions for [Ubuntu](https://github.com/ethereum/go-ethereum/wiki/Installation-Instructions-for-Ubuntu) and it installed version `v1.6.7-stable-ab5646c5` instead of the latest stable release `1.7.3-stable`.

 I build `geth` from source with the latest version and things are now moving forward. 

Also, I tried downloading the latest version from https://geth.ethereum.org/downloads/ and it installed `v1.6.7`.

Bottom line make sure you are using at least 1.7.3 if not build and install from source.

**Comment by holiman** [comment]
> Also, I tried downloading the latest version from https://geth.ethereum.org/downloads/ and it installed v1.6.7.

What? Could  you please elaborate -- what do you see, and what did you click? The page that I see definitely points to 1.7.3 for the latest. Screenshot?

**Comment by pzagor2** [comment]
@holiman  OK. now that I'm checking this, it might just be due to lack of my knowledge.

I download the `geth-linux-386-1.7.3-4bb3c89d.tar.gz` with `wget` and extract it with `tar`. I `cd` in the folder and ran `geth`. But this probably ran the $PATH `geth` and not the one I just extracted.

**Comment by MitchK** [comment]
@pzagor2 I'm using 1.7.3 stable and I compiled it from source.
